core/src/main/java/org/hibernate/ogm/service/impl/OgmIntegrator.java

```diff
@@ -73,10 +72,8 @@ private void attachEventContextManagingListenersIfRequired(SessionFactoryService
 		eventListenerRegistry.addDuplicationStrategy( EventContextManagingFlushEventListenerDuplicationStrategy.INSTANCE );
 		eventListenerRegistry.getEventListenerGroup( EventType.FLUSH ).appendListener( new EventContextManagingFlushEventListener( stateManager ) );
 
-		if ( getIntegrator( JpaIntegrator.class, serviceRegistry ) != null ) {
```

Comment:
I made a commit just for it ;). From what I've seen, the persist operation is used when the key is generated by the datastore.